contracts/DEX/SimpleDex.sol

```diff
@@ -71,4 +88,59 @@ contract SimpleDex {
         uint256 amountOutstanding = order.amount.sub(order.filled);
         return (order, fillAmount, amountOutstanding, order.maker);
     }
-}
+
+    /**
+    * @notice Matches two orders that have already both been made. Has the same
+    *         validation as takeOrder 
+    */
+    function _matchOrder(
+        uint256 order1Id,
+        uint256 order2Id
+    ) internal returns (uint256) {
+
+        // Fill or partially fill order
+        Types.Order storage order1 = orders[order1Id];
+        Types.Order storage order2 = orders[order2Id];
+
+        // Ensure orders can be cancelled against each other
+        require(order1.price == order2.price, "SDX: Price mismatch");
+
+        // Ensure orders are for opposite sides
+        require(order1.side != order2.side, "SDX: Side mismatch");
```

Comment:
Reversion message is slightly misleading as a "mismatch" implies that the sides *don't* match (which is exactly what we want).

Reply:
@sporejack how about "SDX: Same side"?